optimum/intel/openvino/modeling_seq2seq.py

```diff
@@ -310,26 +366,33 @@ class OVEncoder:
             The OpenVINO inference request associated to the encoder.
     """
 
-    def __init__(self, model: openvino.runtime.Model, device: str, ov_config: Dict):
+    def __init__(self, model: openvino.runtime.Model, device: str, ov_config: Dict, main_input_name="input_ids"):
         self.model = model
         self._device = device
         self.device = torch.device("cpu")
         self.input_names = {key.get_any_name(): idx for idx, key in enumerate(self.model.inputs)}
-        self.main_input_name = "input_ids"
+        self.main_input_name = main_input_name
         self.ov_config = ov_config
         self.request = None
 
     @add_start_docstrings_to_model_forward(ENCODER_INPUTS_DOCSTRING)
     def forward(
         self,
-        input_ids: torch.LongTensor,
+        input_ids: torch.LongTensor = None,
         attention_mask: torch.LongTensor = None,
         **kwargs,
     ) -> BaseModelOutput:
         self._compile()
 
         # Model inputs
-        inputs = {"input_ids": input_ids}
+        if input_ids is None and self.main_input_name == "input_ids":
+            raise RuntimeError("input_ids should be provided")
+        elif self.main_input_name != "input_ids":
+            if self.main_input_name not in kwargs:
+                raise RuntimeError(f"{self.main_input_name} should be provided")
+            inputs = {self.main_input_name: kwargs[self.main_input_name]}
+        else:
+            inputs = {"input_ids": input_ids}
```

Comment:
could make sense to have :
```suggestion
        inputs = {self.main_input_name: input_ids}
```

Reply:
looks like it does not work in generate, where encoder called outside of common forward